Sources/Sentry/SentryClient.m

```diff
@@ -638,6 +642,23 @@ - (BOOL)isOOM:(SentryEvent *)event isCrashEvent:(BOOL)isCrashEvent
         [exception.mechanism.type isEqualToString:SentryOutOfMemoryMechanismType];
 }
 
+- (void)storeFreeMemoryToDeviceContext:(SentryEvent *)event
+{
+    if (nil == event.context || event.context.count == 0 || nil == event.context[@"device"]) {
+        return;
+    }
+
+    NSMutableDictionary *context = [[NSMutableDictionary alloc] initWithDictionary:event.context];
+    NSMutableDictionary *device =
+        [[NSMutableDictionary alloc] initWithDictionary:context[@"device"]];
+
+    NSDictionary *systemInfo = [[SentryCrashWrapper sharedInstance] systemInfo];
```

Comment:
With this you still getting cached data from when the SDK was initialized, Also this is what's failing the tests, because systemInfo is being cached before proper configuration of the SDK (a problem in the tests IMO)

I think you should try this instead:
```suggestion
    NSDictionary *systemInfo = SentryCrash.sharedInstance.systemInfo;
```